### ISSUE TITLE
Inconsistent type specification for Buckets [JIRA: CLIENTS-549]

### ISSUE THREAD

**Issue by sata** [basho/riak-erlang-client#226]
-type bucket() :: binary(). %% A bucket name in riakc.hrl

%% @doc Constructor for new riak client objects with an update value and content type.
-spec new(bucket(), key(), value(), content_type()) -> riakc_obj().
new(Bucket, Key, Value, ContentType) ->
    case build_client_object(Bucket, Key, Value) of

But then we have this clause in code: 
build_client_object({T, B0}=B, K, V) when is_binary(T), is_binary(B0), is_binary(K) orelse undefined =:= K ->
    #riakc_obj{bucket = B, key = K, contents = [], updatevalue = V};

In other words, Bucket when using Riak 2 isn't just a binary but it can be a tuple as well.

**Comment by sata** [comment]
This is a duplication of #199

**Comment by Basho-JIRA** [comment]
Fixed, or closed via GitHub issues. 

***[posted via JIRA by Alexander Moore]***
